compiler/noirc_evaluator/src/ssa/ir/instruction.rs

```diff
@@ -847,9 +847,16 @@ fn eval_constant_binary_op(
             if matches!(operator, BinaryOp::Div | BinaryOp::Mod) && rhs == 0 {
                 return None;
             }
-
             let result = function(lhs, rhs);
-            truncate(result, *bit_size).into()
+            if let Some(result) = result {
+                // Check for overflow
+                if result >= 2u128.pow(*bit_size) {
+                    return None;
+                }
+                result.into()
+            } else {
+                return None;
+            }
```

Comment:
```suggestion
            // Check for overflow
            if result >= 2u128.pow(*bit_size) {
                return None;
            }
            result.into()
```

which allows us to remove a layer of nesting here.

Reply:
Done